addons/recoil/functions/fnc_camshake.sqf

```diff
@@ -9,59 +23,52 @@
 #define BASE_FREQ 13
 #define RECOIL_COEF 40
 
-private ["_unit", "_weapon", "_muzzle"];
-
-_unit = _this select 0;
-_weapon = _this select 1;
-_muzzle = _this select 2;
+params ["_unit", "_weapon", "_muzzle"];
 
 if (toLower _weapon in ["throw", "put"]) exitWith {};
 
-private ["_powerMod", "_timeMod", "_freqMod"];
-
-_powerMod = ([0, -0.1, -0.1, 0, -0.2] select (["STAND", "CROUCH", "PRONE", "UNDEFINED", ""] find stance _unit)) + ([0, -1, 0, -1] select (["INTERNAL", "EXTERNAL", "GUNNER", "GROUP"] find cameraView));
-_timeMod = 0;
-_freqMod = 0;
+private _powerMod = ([0, -0.1, -0.1, 0, -0.2] select (["STAND", "CROUCH", "PRONE", "UNDEFINED", ""] find stance _unit)) + ([0, -1, 0, -1] select (["INTERNAL", "EXTERNAL", "GUNNER", "GROUP"] find cameraView));
 
 // to get camshake read kickback
-private ["_config", "_recoil"];
-
-_config = configFile >> "CfgWeapons" >> _weapon;
-_recoil = if (_muzzle == _weapon) then {
-    getText (_config >> "recoil")
-} else {
-    getText (_config >> _muzzle >> "recoil")
-};
+private _recoil = missionNamespace getVariable [format [QGVAR(%1-%2), _weapon, _muzzle], []];
+if (_recoil isEqualTo []) then {
+    private _config = configFile >> "CfgWeapons" >> _weapon;
+    _recoil = if (_muzzle == _weapon) then {
+        getText (_config >> "recoil")
+    } else {
+        getText (_config >> _muzzle >> "recoil")
+    };
 
-if (isClass (configFile >> "CfgRecoils" >> _recoil)) then {
-    _recoil = getArray (configFile >> "CfgRecoils" >> _recoil >> "kickBack");
-    if (count _recoil < 2) then {
+    if (isClass (configFile >> "CfgRecoils" >> _recoil)) then {
+        _recoil = getArray (configFile >> "CfgRecoils" >> _recoil >> "kickBack");
+        if (count _recoil < 2) then {
+            _recoil = [0, 0];
+        };
+    } else {
         _recoil = [0, 0];
     };
-} else {
-    _recoil = [0, 0];
-};
 
-// parse numbers
-_recoil set [0, call compile format ["%1", _recoil select 0]]; 
-_recoil set [1, call compile format ["%1", _recoil select 1]];
+    TRACE_3("Caching Recoil config",_weapon,_muzzle,_recoil);
+
+    // parse numbers
+    _recoil set [0, call compile format ["%1", _recoil select 0]];
```

Comment:
Because these can be either a number or a piece of uncompiled SQF (string). Normally you would read them with getNumber, but here they are the first element inside an array.
parseNumber is no alternative, because it doesn't actually compile the SQF, but stops at the first non-number character it encounters.